adapters/repos/db/migrator.go

```diff
@@ -35,13 +36,28 @@ import (
 	"github.com/weaviate/weaviate/usecases/sharding"
 )
 
+// offloadProvider is an interface has to be implemented by modules
+// to get the backend (s3, azure blob storage, google cloud storage) module
+type offloadProvider interface {
+	// OffloadBackend returns the backend module for (s3, azure blob storage, google cloud storage)
+	OffloadBackend(backend string) (modulecapabilities.OffloadCloud, error)
+}
+
 type Migrator struct {
 	db     *DB
+	cloud  modulecapabilities.OffloadCloud
 	logger logrus.FieldLogger
+	nodeId string
 }
 
-func NewMigrator(db *DB, logger logrus.FieldLogger) *Migrator {
-	return &Migrator{db: db, logger: logger}
+func NewMigrator(db *DB, logger logrus.FieldLogger, provider offloadProvider, nodeName string) (*Migrator, error) {
+	cloud, err := provider.OffloadBackend("offload-s3")
+	if err != nil {
+		// TODO-offload: proper config
+		// here we just return migrator if the offload was not configured
+		return &Migrator{db: db, logger: logger, nodeId: nodeName}, nil
+	}
+	return &Migrator{db: db, logger: logger, cloud: cloud, nodeId: nodeName}, nil
```

Comment:
the offload module is being added to Migrator but is it anywhere used?

Reply:
i will be used in a follow up PR